examples/stable_diffusion_v2/ldm/modules/train/trainer.py

```diff
@@ -148,7 +149,13 @@ def construct(self, *inputs):
                 else:
                     # update LR in each gradient step but not optimize net parameter to ensure the LR curve is
                     # consistent
-                    loss = F.depend(loss, self.optimizer.get_lr())  # .get_lr() will make lr step increased by 1
+                    if self.manually_increase_step:
+                        # for MS >= 2.2, `get_lr`` and `global_step += 1` is decoupled
+                        loss = F.depend(
+                            ops.assign_add(self.optimizer.global_step, self.optimizer.global_step_increase_tensor)
```

Comment:
why use `self.optimizer.global_step_increase_tensor` rather than 1 here?

Reply:
btw, after increase the global step manually, will the LR and loss scale changes accordingly? Can't see from line152 to 155

Reply:
> why use `self.optimizer.global_step_increase_tensor` rather than 1 here?

it follows the internal implementation of Mindspore optimizer, which uses `global_step_increase_tensor` (which is actually just a constant 1), existed in MS2.2

Reply:
> btw, after increase the global step manually, will the LR and loss scale changes accordingly? Can't see from line152 to 155. If so, we can just delete `get_lr` branch for simplicity?

it is the minimum change for compatibility of MS2.1 and MS2.2. If deleting `get_lr` branch, we need to check the intergrity on MS2.1